### PR TITLE
carry addresses within packets and messages

### PR DESCRIPTION
**Pull request by dkmccandless**
Address values accompanying individual message IDs have been transmitted in a separate slice in the envelope, without robust protection of their one-to-one correspondence and semantic relevance. Now that addresses are netip.AddrPort values, it's easier to keep them collocated with their corresponding IDs, within the messages themselves. This simplifies the signatures of many functions involved in encoding, sending, decoding, and receiving.

### CODE REVIEW

File: swim.go [hunk 1]
```diff
@@ -120,10 +120,10 @@ func (n *Node) send(ps []packet) {
 	}
 }
 
-func (n *Node) getAddrs(p packet) (dst netip.AddrPort, addrs []netip.AddrPort) {
+func (n *Node) getAddr(p packet) netip.AddrPort {
 	n.mu.Lock()
 	defer n.mu.Unlock()
-	return n.fsm.getAddrs(p)
+	return n.fsm.getAddr(p)
```
[the remark this answers is not in the document]
Comment (dkmccandless) [reply swim.go:126]: > Also, maybe the packet struct should have an unexported remoteAddr field, analogous to remoteID, and then there's no need for these methods at all?

That's the ticket. That's more appropriate in how it keeps the ID and address together. It'll be great to be able to do that and eliminate this code path completely.

[the remark this answers is not in the document]
Comment (kr) [reply swim.go:126]: Oh sorry! Were you planning on that in a future PR?

[the remark this answers is not in the document]
Comment (dkmccandless) [reply swim.go:126]: I was planning to figure out a way to, but this solution has been right under my nose the entire time. It makes sense to me to do it now.

File: swim.go [hunk 2]
```diff
@@ -105,12 +105,11 @@ func (n *Node) Join(remote netip.AddrPort) {
 
 func (n *Node) send(ps []packet) {
 	for _, p := range ps {
-		dst, addrs := n.getAddrs(p)
-		if dst == (netip.AddrPort{}) {
+		if p.remoteAddr == (netip.AddrPort{}) {
```
Comment (kr) [inline swim.go:108]: Under what circumstances is this possible?

**Review by kr — APPROVED** [review]

File: swim.go (hunk 2 above)
Comment (dkmccandless) [reply swim.go:108]: It shouldn't be; if it is, then there is a bug in the fsm or the way addresses are supplied to it. It seems like a more appropriate behavior in that case would be to panic. If so, then the subsequent call to WriteToUDPAddrPort would return an error, which accomplishes that, and this check can be removed.

Comment (kr) [reply swim.go:108]: Sounds good to me! I agree.

**Pull request merged**